code/modules/mob/living/carbon/xenomorph/castes/slime/abilities_slime.dm

```diff
@@ -0,0 +1,480 @@
+/datum/action/xeno_action/activable/slime_pounce/proc/call_end_ability()
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, PROC_REF(end_ability))
+
+/// Unbuckles the Slime from the living target and punishes them for a successful resist.
+/datum/action/xeno_action/activable/slime_pounce/proc/end_ability()
+	if(!attached_atom)
+		return
+	UnregisterSignal(owner, list(COMSIG_ATOM_EX_ACT, COMSIG_MOB_DEATH, COMSIG_XENOMORPH_FIRE_BURNING, COMSIG_LIVING_DO_RESIST, COMSIG_LIVING_STATUS_STUN, COMSIG_LIVING_STATUS_KNOCKDOWN, COMSIG_LIVING_STATUS_PARALYZE, COMSIG_LIVING_STATUS_UNCONSCIOUS, COMSIG_LIVING_STATUS_SLEEP, COMSIG_LIVING_STATUS_STAGGER))
+	UnregisterSignal(attached_atom, list(COMSIG_PARENT_QDELETING, COMSIG_ATOM_EX_ACT, COMSIG_ATOM_BULLET_ACT))
+	if(isliving(attached_atom))
+		var/mob/living/attached_living = attached_atom
+		UnregisterSignal(attached_living, list(COMSIG_MOVABLE_UNBUCKLE, COMSIG_MOB_DEATH, COMSIG_LIVING_STATUS_UNCONSCIOUS, COMSIG_LIVING_DO_RESIST))
+		if(attached_living.buckled_mobs)
+			attached_living.unbuckle_mob(owner, TRUE)
+			attached_living.remove_movespeed_modifier(MOVESPEED_ID_SLIME_ATTACHED)
+	if(previous_turf)
+		REMOVE_TRAIT(owner, TRAIT_IMMOBILE, TRAIT_GENERIC)
+		owner.anchored = FALSE
+		owner.forceMove(previous_turf)
+		previous_turf = null
+	attached_atom = null
+	owner.layer = MOB_LAYER
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	xeno_owner.Knockdown(SLIME_POUNCE_UNBUCKLE_EFFECT_DURATION)
+	xeno_owner.balloon_alert_to_viewers("Deattached")
+
+// ***************************************
+// *********** Spread Tendrils
+// ***************************************
+/datum/action/xeno_action/spread_tendrils
+	name = "Spread Tendrils"
+	ability_name = "Spread Tendrils"
+	action_icon_state = "spread_tendrils"
+	desc = "Spread hardened, thorn-like tendrils, dealing damage to surrounding enemies and immobilizing them briefly. Increased intensity for victims in the same tile as you."
+	cooldown_timer = 20 SECONDS
+	plasma_cost = 25
+	use_state_flags = XACT_USE_BUCKLED|XACT_USE_CLOSEDTURF
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_SLIME_SPREAD_TENDRILS,
+	)
+
+/datum/action/xeno_action/spread_tendrils/action_activate()
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	if(!do_after(xeno_owner, SLIME_SPREAD_TENDRILS_WINDUP, FALSE, xeno_owner, BUSY_ICON_DANGER))
+		return
+	succeed_activate()
+	add_cooldown()
+	for(var/mob/living/affected_living in range(SLIME_SPREAD_TENDRILS_RANGE, get_turf(xeno_owner)))
+		if(xeno_owner.issamexenohive(affected_living) || affected_living.stat == DEAD)
+			continue
+		var/distance = get_dist(affected_living, xeno_owner)
+		var/damage = xeno_owner.xeno_caste.melee_damage
+		var/effect_duration = SLIME_SPREAD_TENDRILS_EFFECT_DURATION
+		if(distance == 0)
+			damage = damage * SLIME_SPREAD_TENDRILS_CRITICAL_MULTIPLIER
+			effect_duration = effect_duration * SLIME_SPREAD_TENDRILS_CRITICAL_MULTIPLIER
+		if(distance > 1)
```

Comment:
```suggestion
		if(distance == 0)
			damage = damage * SLIME_SPREAD_TENDRILS_CRITICAL_MULTIPLIER
			effect_duration = effect_duration * SLIME_SPREAD_TENDRILS_CRITICAL_MULTIPLIER
		else
```
Only other get_dist is -1 which is a fail state but sincce you use range it will always succeed

Side side note should this be > 1 or >= 1 cus I assume the latter

Reply:
Fair enough, let me do this change.